dbos/dbos.go

```diff
@@ -312,9 +314,35 @@ func NewDBOSContext(inputConfig Config) (DBOSContext, error) {
 	initExecutor.logger.Info("System database initialized")
 
 	// Initialize the queue runner and register DBOS internal queue
-	initExecutor.queueRunner = newQueueRunner()
+	initExecutor.queueRunner = newQueueRunner(initExecutor.logger)
 	NewWorkflowQueue(initExecutor, _DBOS_INTERNAL_QUEUE_NAME)
 
+	// Initialize conductor if API key is provided
+	if config.ConductorAPIKey != "" {
+		if initExecutor.executorID == "local" {
```

Comment:
It's safer to just always generate a new ID here. Otherwise, there's potential for trouble--multiple servers sharing the same ID in particular.